lib/OpenQA/WebAPI/Plugin/ObsRsync/Controller/Folders.pm

```diff
@@ -17,17 +17,47 @@
 package OpenQA::WebAPI::Plugin::ObsRsync::Controller::Folders;
 use Mojo::Base 'Mojolicious::Controller';
 use Mojo::File;
+use POSIX 'strftime';
 
 sub index {
     my $self   = shift;
-    my $folder = $self->param('folder');
     my $helper = $self->obs_rsync;
-    return undef if $helper->check_and_render_error($folder);
-    my $files
-      = Mojo::File->new($helper->home)->list({dir => 1})->grep(sub { -d $_ })->map('basename')
-      ->grep(qr/^(?!t$|profiles$|script$|xml$)/)->to_array;
+    my %folder_info_by_name;
+    my $folders = Mojo::File->new($helper->home)->list({dir => 1})->grep(sub { -d $_ })->map('basename')
+      ->grep(qr/^^(?!t$|profiles$|script$|xml$)/)->to_array;
+
+    for my $folder (@$folders) {
+        my $run_last_info = $helper->get_run_last_info($folder);
+        my ($fail_last_job_id, $fail_last_when) = $helper->get_fail_last_info($folder);
+        $folder_info_by_name{$folder} = {
+            run_last         => $run_last_info->{dt},
+            run_last_version => $run_last_info->{version},
+            run_last_job_id  => $run_last_info->{job_id},
+            fail_last_when   => $fail_last_when,
+            fail_last_job_id => $fail_last_job_id,
+            dirty_status     => $helper->get_dirty_status($folder)};
+    }
 
-    $self->render('ObsRsync_index', folders => $files);
+    my $running_jobs = $self->app->minion->backend->list_jobs(
+        0,
+        4 * $helper->queue_limit,
```

Comment:
I would prefer to not have this magic number. At least you could put it into a constant with a descriptive name

Reply:
I see no benefit from introducing shared constant here as long as you know what that expression is used for. (and the fact that if number of active jobs exceeds $helper->queue_limit, then we are already screwed). Really constant like `FACTOR_FOR_LIMIT_JOBS` makes even less sense than dumb `4`.

Reply:
If `FACTOR_FOR_LIMIT_JOBS_QUERY` is a bad name then don't use it and think of a good name. I don't like magic numbers, too. (And I can't help you with finding a good name because I don't know the meaning of the number.)